core/shared/src/main/scala/zio/prelude/ZValidation.scala

```diff
@@ -287,9 +304,27 @@ sealed trait ZValidation[+W, +E, +A] { self =>
 object ZValidation extends LowPriorityValidationImplicits {
 
   final case class Failure[+W, +E](log: Chunk[W], errors: NonEmptyChunk[E]) extends ZValidation[W, E, Nothing] {
+
+    /** The errors represented in a way which discards the order in which they occurred. */
     lazy val errorsUnordered: NonEmptyMultiSet[E] = NonEmptyMultiSet.fromIterable(errors.head, errors.tail)
+
+    /** The description of the failure. */
+    lazy val message: String =
+      s"errors:\n  ${errors.map(_.toString).mkString("\n  ")}\nlog:\n  ${log.map(_.toString).mkString("\n  ")}"
+
+    /** The same failure, but represented as an `Exception` which also has the `cause` set to the first error that occurred */
+    def toException(implicit ev: E <:< Throwable): Failure.Exception[W, E] = Failure.Exception(this)
   }
-  final case class Success[+W, +A](log: Chunk[W], value: A)                 extends ZValidation[W, Nothing, A]
+
+  object Failure {
+    final case class Exception[+W, +E](failure: Failure[W, E])(implicit ev: E <:< Throwable)
```

Comment:
We're exposing a lot of implementations details here and we are also potentially creating some inaccurate information. The returned `Throwable` is a subtype of `RuntimeException` even though it is possible that none of the original throwable are subtypes of `RuntimeException`. I would suggest we just take the first error and then add the others as suppressed exceptions to that, discarding the log. There are a variety of other operators that users can use to get the log if they want.

Reply:
You mean to mutate an existing Exception by adding others as suppressed? Wouldn't that be a bit misleading? How could users recognize the original suppressed exceptions in that particular Exception from those that we would add?

Or do you think it would be better to use a mere `Exception` instead of `RuntimeException`.

I don't have a strong opinion on this, so let me know, if you still think adding rest as suppressed to the first one is the best way to go.